src/components/Aframe/my-loader.js

```diff
@@ -432,141 +432,85 @@ AFRAME.registerComponent("myloader", {
   },
 
   updateColorMapping: function () {
-    var imgColorImage = document.querySelector(".colorMapImg");
-    var imgWidth = imgColorImage.width;
-    var imgHeight = imgColorImage.height;
-    var colorCanvas = document.createElement("canvas");
-    // var el = this.el;
-
-    // var alpha = this.alphaData;
-    var colorTransfer = this.colorTransfer;
-    var iam = this;
-    this.colorMap.img.onload = function (data) {
-      colorCanvas.height = imgHeight;
-      colorCanvas.width = imgWidth;
-      var colorContext = colorCanvas.getContext("2d");
-      colorContext.drawImage(imgColorImage, 0, 0);
-      var colorData = colorContext.getImageData(0, 0, imgWidth, 1).data;
-      colorTransfer = new Uint8Array(3 * 256);
-      for (var i = 0; i < 256; i++) {
-        colorTransfer[i * 3] = colorData[i * 4];
-        colorTransfer[i * 3 + 1] = colorData[i * 4 + 1];
-        colorTransfer[i * 3 + 2] = colorData[i * 4 + 2];
-      }
-      iam.colorTransfer = colorTransfer;
-      iam.updateTransfertexture();
-    };
-    this.colorMap.img.src = imgColorImage.src;
+    if (!this.colorTransferMap.has(this.currentColorMap)) {
+      var colorCanvas = document.createElement("canvas");
+      var iam = this;
+      let newColorMap = {
+        img: document.createElement("img"),
+        width: 255,
+        height: 15,
+        data: null,
+      };
+      var imgWidth = newColorMap.width;
+      var imgHeight = newColorMap.height;
+
+      newColorMap.img.src = this.currentColorMap;
+      this.colorTransferMap.set(this.currentColorMap, newColorMap);
+      const mappedColorMap = newColorMap;
```

Comment:
why create this pointer?

Reply:
I am hesitant on passing the colorTransferMap to the promise call. Dont want to change this part at this moment.